integration-tests/src/tests/client/features/zero_balance_account.rs

```diff
@@ -0,0 +1,249 @@
+use assert_matches::assert_matches;
+
+use near_chain::ChainGenesis;
+use near_chain_configs::Genesis;
+use near_client::adapter::ProcessTxResponse;
+use near_client::test_utils::TestEnv;
+use near_crypto::{InMemorySigner, KeyType, PublicKey};
+use near_primitives::account::id::AccountId;
+use near_primitives::errors::{ActionError, ActionErrorKind, InvalidTxError, TxExecutionError};
+use near_primitives::shard_layout::ShardUId;
+use near_primitives::transaction::{AddKeyAction, SignedTransaction};
+use near_primitives::version::ProtocolFeature;
+use near_primitives::views::{FinalExecutionStatus, QueryRequest, QueryResponseKind};
+use nearcore::config::GenesisExt;
+
+use crate::tests::client::runtimes::create_nightshade_runtimes;
+use near_primitives::account::AccessKey;
+use near_primitives::config::ActionCosts;
+use near_primitives::runtime::config::RuntimeConfig;
+use near_primitives::transaction::Action::AddKey;
+use near_primitives::types::Balance;
+
+/// Test 2 things: 1) a valid zero balance account can be created and 2) a nonzero balance account
+/// (one with a contract deployed) cannot be created without maintaining an initial balance
+#[cfg(feature = "nightly_protocol")]
+#[test]
+fn test_zero_balance_account_creation() {
+    let epoch_length = 1000;
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+    genesis.config.epoch_length = epoch_length;
+    genesis.config.protocol_version = ProtocolFeature::ZeroBalanceAccount.protocol_version();
+    let mut env = TestEnv::builder(ChainGenesis::test())
+        .runtime_adapters(create_nightshade_runtimes(&genesis, 1))
+        .build();
+    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+
+    let new_account_id: AccountId = "hello.test0".parse().unwrap();
+    let signer0 = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let new_signer =
+        InMemorySigner::from_seed(new_account_id.clone(), KeyType::ED25519, "hello.test0");
+
+    // create a valid zero balance account. Transaction should succeed
+    let create_account_tx = SignedTransaction::create_account(
+        1,
+        signer0.account_id.clone(),
+        new_account_id.clone(),
+        0,
```

Comment:
consider enabling this whole test right now - and having the condition here -- if nightly -> pass 0 as balance, if not pass something else.

Reply:
I don't think this is necessary as 1) it would make the test more complex and harder to read and 2) we already test the current behavior in other runtime tests such as `test_create_account_failure_no_funds` and `test_validate_transaction_invalid_low_balance` (which were modified) to make sure that the zero balance account does not interfere with the stable version